### PR TITLE
fix EventStreamSpec instability

### LINKED ISSUE #30460: EventStreamSpec: unsubscribe the actor, when it subscribes already in terminated state
https://github.com/akka/akka/pull/30449/checks?check_run_id=3233378543

```
[info] - must unsubscribe the actor, when it subscribes already in terminated state *** FAILED *** (56 milliseconds)
[info]   false did not equal true (EventStreamSpec.scala:339)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.matchers.MatchersHelper$.indicateFailure(MatchersHelper.scala:402)
[info]   at org.scalatest.matchers.should.Matchers.should(Matchers.scala:6924)
[info]   at org.scalatest.matchers.should.Matchers.should$(Matchers.scala:1808)
[info]   at akka.testkit.AkkaSpec.should(AkkaSpec.scala:54)
[info]   at akka.event.EventStreamSpec.f$proxy17$1(EventStreamSpec.scala:339)
[info]   at akka.event.EventStreamSpec.$init$$$anonfun$15$$anonfun$14(EventStreamSpec.scala:318)```

### PR DESCRIPTION
**Pull request by raboof**
fixes #30460

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/event/EventStreamSpec.scala [hunk 1]
```diff
@@ -337,13 +342,26 @@ class EventStreamSpec extends AkkaSpec(EventStreamSpec.config) {
         es.subscribe(target, classOf[A]) should ===(true)
         fishForDebugMessage(a2, s"unsubscribing $target from all channels")
 
-        es.subscribe(target, classOf[A]) should ===(true)
-        fishForDebugMessage(a2, s"unsubscribing $target from all channels")
+        after(30.millis) {
```
Comment (andreaTP) [inline akka-actor-tests/src/test/scala/akka/event/EventStreamSpec.scala:345]: don't we have a more deterministic event to wait for? Or can we wrap the block into an `awaitAssert`?

Comment (raboof) [reply akka-actor-tests/src/test/scala/akka/event/EventStreamSpec.scala:345]: > don't we have a more deterministic event to wait for?

I don't think so

> Or can we wrap the block into an `awaitAssert`?

I'm not sure that's better?

Comment (andreaTP) [reply akka-actor-tests/src/test/scala/akka/event/EventStreamSpec.scala:345]: > I'm not sure that's better?

well, a random timeout might always fail depending on the machine running the test, `awaitAssert` should be a little more robust ...

Comment (raboof) [reply akka-actor-tests/src/test/scala/akka/event/EventStreamSpec.scala:345]: Ah, I guess we could put *just* the `subscribe` (but not the `fishForDebugMessage`) in the `awaitAssert`, I'm OK with that. Will update.

**Review by andreaTP — APPROVED** [review]

**Pull request merged**